src/network/networkd-link.c

```diff
@@ -689,7 +689,9 @@ static Address* link_find_dhcp_server_address(Link *link) {
 static int link_enter_configured(Link *link) {
         assert(link);
         assert(link->network);
-        assert(link->state == LINK_STATE_SETTING_ROUTES);
+
+        if (link->state != LINK_STATE_SETTING_ROUTES)
+                return 1;
```

Comment:
It is *also* used as a callback. The return convention is determined by this use.